### ISSUE TITLE
World refresh on save as

### ISSUE THREAD

**Issue by Eiyeron** [bjorn/tiled#2033]
Hello, I'm currently trying out the world feature and I love it, it's pretty dang useful to make a map layout based on older limitations. But I'm facing a small workflow issue : what's the best way to make the world layout update when I create a new map.

Currently my process is :
- save a current map as a new one with new coordinates
- (empty the new map)
- unload the world
- load the world

I guess it's fine when you a few amount of maps or small maps, but I wonder if it won't get problematic in the future. Thus I was wondering if detecting if the world must be updated if the saved-as map matches the world would make a good feature.

Thanks and have a nice day.

Note : I'm using regex matching to automatically place maps, if that can help

**Comment by bjorn** [comment]
> Note : I'm using regex matching to automatically place maps, if that can help

Right, this issue only applies in that case, because in manually listed maps you'd have to add the map to the list, and Tiled will detect this change and reload the world automatically. In your case, a workaround that may be faster than unloading/loading the world would be to touch / save the world file.

But, for sure it would be nice if this could happen automatically in that case. We could indeed check if a map belongs to a loaded world after saving it. Alternatively, we could watch the directory where the world file resides for changes, so that we can also add / remove maps from the world when those files are changed externally.
